File: dotnet-stack/backend/Controllers/AuthController.cs
using Microsoft.AspNetCore.Authorization;
using Microsoft.AspNetCore.Mvc;
using WildernessStays.Core.Contracts;
using WildernessStays.Core.Services;

namespace WildernessStays.Api.Controllers;

[ApiController]
[Route("api/auth")]
public class AuthController(AuthService auth) : ControllerBase
{
    [HttpPost("register")]
    public async Task<IActionResult> Register(RegisterDto dto) =>
        StatusCode(201, await auth.RegisterAsync(dto.Name, dto.Email, dto.Password));

    [HttpPost("login")]
    public async Task<IActionResult> Login(LoginDto dto) =>
        StatusCode(201, await auth.LoginAsync(dto.Email, dto.Password));

    [HttpGet("me")]
    [Authorize]
    public IActionResult Me() => Ok(new
    {
        id = User.FindFirst("sub")?.Value,
        name = User.FindFirst("name")?.Value,
        email = User.FindFirst("email")?.Value,
        role = User.FindFirst("role")?.Value,
    });
}

File: node-stack/backend/src/auth/auth.controller.ts
import { Body, Controller, Get, Post, UseGuards } from '@nestjs/common';
import { AuthService } from './auth.service';
import { AuthGuard, CurrentUser } from './auth.guard';
import { LoginDto, RegisterDto } from './dto/auth.dto';
import type { JwtPayload } from './auth.service';

@Controller('auth')
export class AuthController {
  constructor(private readonly auth: AuthService) {}

  @Post('register')
  register(@Body() dto: RegisterDto) {
    return this.auth.register(dto.name, dto.email, dto.password);
  }

  @Post('login')
  login(@Body() dto: LoginDto) {
    return this.auth.login(dto.email, dto.password);
  }

  @Get('me')
  @UseGuards(AuthGuard)
  me(@CurrentUser() user: JwtPayload) {
    return { id: user.sub, name: user.name, email: user.email, role: user.role };
  }
}
